types.go

```diff
@@ -16,19 +16,21 @@ package merger
 
 import (
 	"context"
+	"io"
 	"io/ioutil"
 	"time"
 
 	"github.com/dfuse-io/dstore"
 )
 
 type OneBlockFile struct {
-	name       string
-	blockTime  time.Time
-	id         string
-	num        uint64
-	previousID string
-	data       []byte
+	canonicalName string
+	filenames     []string
```

Comment:
Based on the usage pattern I saw, I think a `map` would fit better here.
```suggestion
	filenames     map[string]bool
```